clang/test/Driver/sycl-offload.c

```diff
@@ -762,80 +739,46 @@
 // CHK-PHASE-MULTI-TARG: 10: input, "[[INPUT]]", c, (device-sycl)
 // CHK-PHASE-MULTI-TARG: 11: preprocessor, {10}, cpp-output, (device-sycl)
 // CHK-PHASE-MULTI-TARG: 12: compiler, {11}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 13: backend, {12}, assembler, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 14: assembler, {13}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 15: linker, {14}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 16: llvm-spirv, {15}, spirv, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 17: clang-offload-wrapper, {16}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 18: input, "[[INPUT]]", c, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 19: preprocessor, {18}, cpp-output, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 20: compiler, {19}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 21: backend, {20}, assembler, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 22: assembler, {21}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 23: linker, {22}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 24: llvm-spirv, {23}, spirv, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 25: backend-compiler, {24}, fpga_aocx, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 26: clang-offload-wrapper, {25}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 27: compiler, {3}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 28: backend, {27}, assembler, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 29: assembler, {28}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 30: linker, {29}, ir, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 31: llvm-spirv, {30}, spirv, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 32: backend-compiler, {31}, image, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 33: clang-offload-wrapper, {32}, object, (device-sycl)
-// CHK-PHASE-MULTI-TARG: 34: offload, "host-sycl (x86_64-unknown-linux-gnu)" {9}, "device-sycl (spir64-unknown-unknown-sycldevice)" {17}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {26}, "device-sycl (spir64_gen-unknown-unknown-sycldevice)" {33}, image
-
-/// ###########################################################################
-
-/// offload with multiple AOT targets & additional AOT binaries passed through fsycl-add-targets
```

Comment:
NB: This test case was removed altogether due to its practical non-usability.

Ending up with a couple of AOT-compiled binaries for the same target in the fat binary would imply that the offline device code linking didn't observe all of the translation units. This could lead to unexpected behavior (kernels' missing/having different definitions in different images), the image selection mechanism not being hard-wired to a particular device binary of the 2.